examples/models/file/latex_extension_mathjax.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+from scipy.special import jv
+
+from bokeh.models import Label
+from bokeh.palettes import Spectral4
+from bokeh.core.properties import (
+  Nullable,
+  Float
+)
+from bokeh.plotting import figure, output_file, show
+from bokeh.util.compiler import TypeScript
+
+output_file('latex_extension_with_mathjax.html')
+
+class LatexLabel(Label):
+    width = Nullable(Float, help="""
+    The gradient of the line, in data units
+    """)
+    height = Nullable(Float, help="""
+    The gradient of the line, in data units
+    """)
+    """A subclass of `Label` with all of the same class attributes except
+    canvas mode isn't supported and DOM manipulation happens in the TypeScript
+    superclass implementation that requires setting `render_mode='css'`).
+
+    Only the render method of LabelView is overwritten to perform the
+    text -> latex (via MathJax) conversion
+    """
+    __javascript__ = ["https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"]
+    __implementation__ = TypeScript("""
+import {Label, LabelView} from "models/annotations/label"
+import * as p from "core/properties"
+
+declare namespace MathJax {
+  function tex2svg(input: string): any
+}
+
+export class LatexLabelView extends LabelView {
+  model: LatexLabel
+
+  protected _render(): void {
+    const panel = this.layout != null ? this.layout : this.plot_view.frame
+
+    const xscale = this.coordinates.x_scale
+    const yscale = this.coordinates.y_scale
+
+    let sx = this.model.x_units == "data" ? xscale.compute(this.model.x) : panel.bbox.xview.compute(this.model.x)
+    let sy = this.model.y_units == "data" ? yscale.compute(this.model.y) : panel.bbox.yview.compute(this.model.y)
+
+    sx += this.model.x_offset
+    sy -= this.model.y_offset
+
+    const svgElement = MathJax.tex2svg(this.model.text).children[0]
+
+    let outerHTML = svgElement.outerHTML,
+      blob = new Blob([outerHTML],{type:'image/svg+xml;charset=utf-8'});
+
+    let URL = window.URL || window.webkitURL || window;
```

Comment:
Is this really necessary? It seems that all browsers we support (including IE), support `URL.createObjectURL()`. Also, which browser/spec allows for access through `window`?

Reply:
Additionally, you shouldn't access globals through `window`, because it's redundant and this code may run (in future) in an environment where the global context is not `window` (e.g. a web worker global context). If you need to check if an object exists globally, like this `URL`, then use `typeof URL !== "undefined"`.